syntax/forth.vim

```diff
@@ -181,36 +180,28 @@ syn keyword forthBlocks BLOCK-INCLUDED BLK
 " numbers
 syn keyword forthMath DECIMAL HEX BASE
 syn match forthInteger '\<-\=[0-9]\+.\=\>'
-syn match forthInteger '\<&-\=[0-9]\+.\=\>'
 syn match forthInteger '\<#-\=[0-9]\+.\=\>'
-" recognize hex and binary numbers, the '$' and '%' notation is for gforth
-syn match forthInteger '\<\$\x*\x\+\>' " *1* --- don't mess
-syn match forthInteger '\<\x*\d\x*\>'  " *2* --- this order!
-syn match forthInteger '\<%[0-1]*[0-1]\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe]\d\+\>'
-syn match forthFloat '\<-\=\d*[.]\=\d\+[DdEe][-+]\d\+\>'
+syn match forthInteger '\<\$-\=\x*\x\+\>'
+syn match forthInteger '\<%-\=[0-1]*[0-1]\+\>'
+syn match forthFloat '\<[+-]\=\d*[.]\=\d*[DdEe][+-]\=\d*\>'
 
 " XXX If you find this overkill you can remove it. this has to come after the
 " highlighting for numbers otherwise it has no effect.
 syn region forthComment start='0 \[if\]' end='\[endif\]' end='\[then\]' contains=forthTodo
 
 " Strings
-syn region forthString start=+\.*"+ end=+"+ end=+$+ contains=@Spell
-" XXX
-syn region forthString start=+s"+ end=+"+ end=+$+ contains=@Spell
-syn region forthString start=+c"+ end=+"+ end=+$+ contains=@Spell
-
+syn region forthString matchgroup=forthString start=+s"\s+ end=+"+ end=+$+ contains=@Spell
+syn region forthString matchgroup=forthString start=+c"\s+ end=+"+ end=+$+ contains=@Spell
+syn region forthString matchgroup=forthString start=+\."\s+ end=+"+ end=+$+ contains=@Spell
 syn region forthString matchgroup=forthString start=+s\\"+ end=+"+ end=+$+ contains=@Spell,forthEscape
 
 syn match forthEscape +\\[abeflmnqrtvz"\\]+ contained
 syn match forthEscape "\\x\x\x" contained
 
 " Comments
 syn match forthComment '\\\%(\s.*\)\=$' contains=@Spell,forthTodo,forthSpaceError
-syn region forthComment start='\\S\s' end='.*' contains=@Spell,forthTodo,forthSpaceError
 syn match forthComment '\.(\s[^)]*)' contains=@Spell,forthTodo,forthSpaceError
 syn region forthComment start='\(^\|\s\)\zs(\s' skip='\\)' end=')' contains=@Spell,forthTodo,forthSpaceError
-syn region forthComment start='/\*' end='\*/' contains=@Spell,forthTodo,forthSpaceError
```

Comment:
`/* ... */` definitely does not look like Forth :-)

Reply:
Like `\S` I've seen it used but it's not common and definitely not standard.